airflow-core/src/airflow/ui/src/pages/Pools/Pools.tsx

```diff
@@ -32,8 +33,17 @@ export const Pools = () => {
   const [searchParams, setSearchParams] = useSearchParams();
   const { NAME_PATTERN: NAME_PATTERN_PARAM }: SearchParamsKeysType = SearchParamsKeys;
   const [poolNamePattern, setPoolNamePattern] = useState(searchParams.get(NAME_PATTERN_PARAM) ?? undefined);
+  
+  const { tableURLState } = useTableURLState();
```

Comment:
Thanks for picking this up.

We need `setTableURLState` from here too and pass it to the table component like so: `<DataTable onStateChange={setTableURLState}`.

Reply:
Added @bbovenzi. Please let me know if more is needed. Thank you for your time